File: CPlusPlus/Adapter/src/Center.cpp
/*************************************************************************
    > File Name: Center.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年02月28日 星期二 14时29分03秒
 ************************************************************************/

#include"Center.h"
#include<string>
#include<iostream>

using namespace std;

Center::Center(const string& str):name(str)
{
	;
}

Center::~Center()
{
	;
}

void Center::Attack()
{
	cout<<"Center Attack!"<<endl;
}

void Center::Defence()
{
	cout<<"Center Defence!"<<endl;
}

File: Python/Adapter/Center.py
# -*- coding:utf-8 -*-
from ChineseCenter import *

class Center:

    def __init__(self, str):
        self.__name = str

    def Attack(self):
        print "Center Attack!"

    def Defence(self):
        print "Center Defence!"

class CenterTranslator(Center):

    def __init__(self, str):
        Center.__init__(self, str)
        self.chiCenter = ChineseCenter(str)

    def Attack(self):
        self.chiCenter.ChineseAttack()
    
    def Defence(self):
        self.chiCenter.ChineseDefence()





